bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairControllerHandler.java

```diff
@@ -0,0 +1,993 @@
+
+        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            goOffline(ThingStatusDetail.BRIDGE_OFFLINE);
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            goOnline();
+        }
+    }
+
+    public void goOnline() {
+        logger.debug("Thing {} goOnline.", getThing().getUID());
+
+        this.waitStatusForOnline = false;
+
+        // make sure bridge exists and is online
+        PentairBaseBridgeHandler bh = getBridgeHandler();
+        if (bh == null) {
+            logger.error("Bridge does not exist");
+            return;
+        }
+
+        if (onlineController != null) {
+            logger.debug("Another controller is already configured");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Another controller is already configured.");
+        }
+
+        ThingStatus ts = bh.getThing().getStatus();
+        if (!ts.equals(ThingStatus.ONLINE)) {
+            logger.debug("Bridge is not online");
+            return;
+        }
+
+        waitStatusForOnline = true; // Wait for first status response to go online
+    }
+
+    public void finishOnline() {
+        onlineController = this;
+        updateStatus(ThingStatus.ONLINE);
+
+        // setup timer to sync time
+        Runnable runnable = new Runnable() {
+
+            @Override
+            public void run() {
+                boolean synctime = ((boolean) getConfig().get("synctime"));
+                if (synctime) {
+                    logger.info("Synchronizing System Time");
+                    Calendar now = Calendar.getInstance();
+                    setClockSettings(now.get(Calendar.HOUR_OF_DAY), now.get(Calendar.MINUTE),
+                            now.get(Calendar.DAY_OF_WEEK), now.get(Calendar.DAY_OF_MONTH), now.get(Calendar.MONTH) + 1,
+                            now.get(Calendar.YEAR) - 2000);
+                }
+            }
+        };
+
+        // setup syncTimeJob to run once a day, initial time to sync is 3 minutes after controller goes online. This is
+        // to prevent collision with main thread queries on initial startup
+        syncTimeJob = scheduler.scheduleWithFixedDelay(runnable, 3, 24 * 60 * 60, TimeUnit.MINUTES);
+
+        Runnable queryinfo = new Runnable() {
+            @Override
+            public void run() {
+                int i;
+
+                getSWVersion();
+
+                getHeat();
+
+                getClockSettings();
+
+                for (i = 1; i <= NUMCIRCUITS; i++) {
+                    getCircuitNameFunction(i);
+                }
+
+                for (i = 1; i <= NUMSCHEDULES; i++) {
+                    getSchedule(i);
+                }
+
+                getLightGroups();
+
+                getValves();
+
+                getSWVersion();
+            }
+        };
+        Thread thread = new Thread(queryinfo);
+        thread.start();
```

Comment:
This thread is only called once and does not need to repeat so i didn't think scheduleWithFixedDelay was the right method.  The reason a separate thread is required is because this is a called as part of a synchronous callback from the parser thread.  Since this function (finishOnline) then sends additional messages and expects additional packets in return, the parser can't stall.

Reply:
From resource and code perspective it's better to let the framework do the thread handling. You can use `schedule()` or `execute()` for one-shot tasks.